plugins/common/utils.go

```diff
@@ -124,3 +134,71 @@ func getCiValue() bool {
 	}
 	return ci
 }
+
+// Get project key from flag or environment variable
+func GetProject(c *components.Context) string {
+	projectKey := c.GetStringFlagValue("project")
+	return getOrDefaultEnv(projectKey, coreutils.Project)
+}
+
+// Return argument if not empty or retrieve from environment variable
+func getOrDefaultEnv(arg, envKey string) string {
+	if arg != "" {
+		return arg
+	}
+	return os.Getenv(envKey)
+}
+
+func CreateDownloadConfiguration(c *components.Context) (downloadConfiguration *artifactoryUtils.DownloadConfiguration, err error) {
```

Comment:
Some of these aren't really plugins utils, I suggest placing them in jfrog-cli-artifactory.
In addition, since they duplicate existing functions in `jfrog-cli` (only differ in context), let's try to make a common function out of them or somehow overcome the context difference

Reply:
Agree moved lifecycle functions to `jfrog-cli-artifactory`